src/Components/WebAssembly/WebAssembly.Authentication/src/Services/AuthorizationMessageHandler.cs

```diff
@@ -0,0 +1,124 @@
+            }
+
+            if (_allowedUris.Any(uri => uri.IsBaseOf(request.RequestUri)))
+            {
+                if (_lastToken == null || now >= _lastToken.Expires.AddMinutes(-5))
+                {
+                    var tokenResult = _tokenOptions != null ?
+                        await _provider.RequestAccessToken(_tokenOptions) :
+                        await _provider.RequestAccessToken();
+
+                    if (tokenResult.TryGetToken(out var token))
+                    {
+                        _lastToken = token;
+                        _cachedHeader = new AuthenticationHeaderValue("Bearer", _lastToken.Value);
+                    }
+                    else
+                    {
+                        throw new AccessTokenNotAvailableException(tokenResult, _navigation);
+                    }
+                }
+
+                // We don't try to handle 401s and retry the request with a new token automatically since that would mean we need to copy the request
+                // headers and buffer the body and we expect that the user instead handles the 401s. (Also, we can't really handle all 401s as we might
+                // not be able to provision a token without user interaction).
+                request.Headers.Authorization = _cachedHeader;
+            }
+
+            return await base.SendAsync(request, cancellationToken);
+        }
+
+        /// <summary>
+        /// Configures this handler to authorize outbound HTTP requests using an access token. The access token is only attached if only attached if at least one of
+        /// <paramref name="endpointUrls" /> is a base of <see cref="HttpRequestMessage.RequestUri" />.
+        /// </summary>
+        /// <param name="endpointUrls">The base addresses of endpoint URLs to which the token will be attached.</param>
+        /// <param name="scopes">The list of scopes to use when requesting an access token.</param>
+        /// <param name="returnUrl">The return URL to use in case there is an issue provisioning the token and a redirection to the
+        /// identity provider is necessary.
+        /// </param>
+        /// <returns>This <see cref="AuthorizationMessageHandler"/>.</returns>
+        public AuthorizationMessageHandler ConfigureHandler(
+            IEnumerable<string> endpointUrls,
+            IEnumerable<string> scopes = null,
+            string returnUrl = null)
+        {
+            if (_allowedUris != null)
+            {
+                throw new InvalidOperationException("Handler already configured.");
+            }
+
+            if (endpointUrls == null)
+            {
+                throw new ArgumentNullException(nameof(endpointUrls));
+            }
+
+            var uris = endpointUrls.Select(uri => new Uri(uri, UriKind.Absolute)).ToArray();
+            if (uris.Length == 0)
+            {
+                throw new ArgumentException("At least one uri must be configured.", nameof(endpointUrls));
```

Comment:
```suggestion
                throw new ArgumentException("At least one URL must be configured.", nameof(endpointUrls));
```

We're not super consistent about the terminology URL vs URI overall in ASP.NET, but it's good to be consistent within a single line at least :)